site/en/docs/android/custom-tabs/howto-custom-tab-native-apps/index.md

```diff
@@ -0,0 +1,114 @@
+Android to avoid browsers when launching. If a native app that is capable of handling this Intent is not found, an
+`ActivityNotFoundException` will be thrown.
+
+```java
+static boolean launchNativeApi30(Context context, Uri uri) {
+    Intent nativeAppIntent = new Intent(Intent.ACTION_VIEW, uri)
+            .addCategory(Intent.CATEGORY_BROWSABLE)
+            .addFlags(Intent.FLAG_ACTIVITY_NEW_TASK |
+                    Intent.FLAG_ACTIVITY_REQUIRE_NON_BROWSER);
+    try {
+        context.startActivity(nativeAppIntent);
+        return true;
+    } catch (ActivityNotFoundException ex) {
+        return false;
+    }
+}
+```
+
+## Before Android 11
+Even though the application may target Android 11, or API level 30, previous Android versions will
+not understand the `FLAG_ACTIVITY_REQUIRE_NON_BROWSER` flag, so we need to resort to querying the
+Package Manager in those cases:
+
+```java
+private static boolean launchNativeBeforeApi30(Context context, Uri uri) {
+    PackageManager pm = context.getPackageManager();
+
+    // Get all Apps that resolve a generic url
+    Intent browserActivityIntent = new Intent()
+            .setAction(Intent.ACTION_VIEW)
+            .addCategory(Intent.CATEGORY_BROWSABLE)
+            .setData(Uri.fromParts("http", "", null));
+    Set<String> genericResolvedList = extractPackageNames(
+            pm.queryIntentActivities(browserActivityIntent, 0));
+
+    // Get all apps that resolve the specific Url
+    Intent specializedActivityIntent = new Intent(Intent.ACTION_VIEW, uri)
+            .addCategory(Intent.CATEGORY_BROWSABLE);
+    Set<String> resolvedSpecializedList = extractPackageNames(
+            pm.queryIntentActivities(specializedActivityIntent, 0));
+
+    // Keep only the Urls that resolve the specific, but not the generic
+    // urls.
+    resolvedSpecializedList.removeAll(genericResolvedList);
+
+    // If the list is empty, no native app handlers were found.
+    if (resolvedSpecializedList.isEmpty()) {
+        return false;
+    }
+
+    // We found native handlers. Launch the Intent.
+    specializedActivityIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+    context.startActivity(specializedActivityIntent);
+    return true;
+}
+```
+
+The approach used here is to query the Package Manager for applications that support a generic
+"http" intent. Those applications are likely browsers.
+
+Then, query for applications that handle intents for the specific URL we want to launch. This will
+return both browsers and native applications setup to handle that URL.
+
+Now, remove all browsers found on the first list from the second list, and we'll be left only with
+native apps.
+
+If the list is empty, we know there are no native handlers and return false. Otherwise, we launch
+the intent for the native handler.
+
+### Putting it all together
+
+We need to ensure using the right method for each occasion: 
```

Comment:
I feel like there is a noun missing in this sentence. "__we are__ using", maybe?